cmd/controller/main.go

```diff
@@ -140,7 +138,7 @@ func main() {
 	configurationInformer := servingInformerFactory.Serving().V1alpha1().Configurations()
 	revisionInformer := servingInformerFactory.Serving().V1alpha1().Revisions()
 	kpaInformer := servingInformerFactory.Autoscaling().V1alpha1().PodAutoscalers()
-	buildInformer := buildInformerFactory.Build().V1alpha1().Builds()
+	buildInformer := sharedInformerFactory.Duck().V1alpha1().KResources()
```

Comment:
FWIW, I think this is only going to look at resources under the `duck.knative.dev/v1alpha1` `apiVersion` and only `KResources` at that.

There is actually an item on my TODO list to figure out how to reduce the amount of codegen for `duck:v1alpha1`, which will possibly even eliminate the `KResource` bits from the shared client.

Reply:
FWIW, this is one bit that I expect to be hard.  You might consider splitting this further into:
1a) Build creation  -- how you are currently using the dynamic client.
1b) Build tracking  -- what we currently use the Build informer for.

Reply:
> There is actually an item on my TODO list to figure out how to reduce the amount of codegen for `duck:v1alpha1`, which will possibly even eliminate the `KResource` bits from the shared client.

Is something like HasConditions coming?

Reply:
I think @n3wscott is interested in that, but we don't yet.